VimR/Workspace/Workspace.swift

```diff
@@ -158,20 +169,34 @@ extension Workspace {
   }
 
   override func draggingExited(_ sender: NSDraggingInfo?) {
-    self.endDrag()
+    self.endDrag(sender: sender)
   }
 
   override func draggingEnded(_ sender: NSDraggingInfo?) {
-    self.endDrag()
+    self.endDrag(sender: sender)
   }
 
-  fileprivate func endDrag() {
+  fileprivate func endDrag(sender: NSDraggingInfo?) {
     self.isDragOngoing = false
+    if isFile(sender: sender) {
+        return
+    }
     self.draggedOnBarLocation = nil
     self.proxyBar.removeFromSuperview()
   }
 
+  fileprivate func isFile(sender: NSDraggingInfo?) -> Bool! {
```

Comment:
I think that (especially implicit) optionals should be avoided if possible. Since in `draggingUpdated()` `NSDraggingInfo` object is not optional:
```
fileprivate func isFile(sender: NSDraggingInfo) -> Bool {
  return (... some optional bool ...) ?? false
}
```